tests/unittests/benchmark/test_benchmark_client.py

```diff
@@ -306,3 +307,60 @@ def insert_race_condition(*args, **kwargs):
 
             assert bm.configuration == benchmark_config
             assert count_benchmarks() == 1
+
+    def test_create_with_executor(self, benchmark_config, benchmark_config_py):
+
+        with OrionState():
+            config = copy.deepcopy(benchmark_config_py)
+            bm1 = get_or_create_benchmark(**config)
+
+            assert bm1.configuration == benchmark_config
+            assert bm1.executor.n_workers == orion.core.config.worker.n_workers
+
```

Comment:
Maybe ensure that orion.core.config.worker.n_workers != 2 so that we can catch is n_workers is not overridden correctly by Joblib executor.